tensorrt_llm/_torch/auto_deploy/distributed/common.py

```
@@ -131,25 +184,53 @@ def initialize(rank: int = 0, world_size: int = 1, port: Optional[int] = None) -
+                    port = get_free_port()
+            else:
+                # All retries exhausted
+                shared_port.value = -1  # Signal failure
+                port_ready_barrier.wait()
+                raise RuntimeError(f"Failed to find available port after {max_retries} attempts")
+        else:
+            # Other ranks: wait for rank 0 to find a working port
+            port_ready_barrier.wait()
+            port = shared_port.value
+            if port == -1:
+                raise RuntimeError("Rank 0 failed to initialize, cannot proceed")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)